gammapy/datasets/map.py

```diff
@@ -2694,6 +2729,36 @@ def to_map_dataset(self, name=None):
             meta_table=self.meta_table,
         )
 
+    def _to_asimov_dataset(self, name=None, models=None):
+        """Create Asimov dataset"""
+        npred = self.npred()
+        data = np.nan_to_num(npred.data, copy=True, nan=0.0, posinf=0.0, neginf=0.0)
+        npred.data = data.astype("float")
+
+        npred_off = get_wstat_mu_bkg(
+            n_on=npred.data,
+            n_off=self.counts_off.data,
+            alpha=self.alpha.data,
+            mu_sig=npred.data,
+        )
+        npred_off = Map.from_geom(geom=self._geom, data=npred_off)
+
+        return self.__class__(
+            models=models,
+            counts=npred,
+            counts_off=npred_off,
+            exposure=self.exposure,
+            acceptance=self.acceptance,
+            acceptance_off=self.acceptance_off,
+            psf=self.psf,
+            edisp=self.edisp,
+            mask_safe=self.mask_safe,
+            mask_fit=self.mask_fit,
+            gti=self.gti,
+            name=name,
+            meta=self.meta,
```

Comment:
Maybe `MapDataset.copy()` can be adjusted like `Observation.copy` with an `in_memory` option. In a separate PR

Reply:
I think the `in_memory` option of `Observation.copy` has a different purpose (overwrite the lazy fits data loading)

Reply:
could add a `_init_copy` method in separate PR.